pkg/semrel/semrel_test.go

```diff
@@ -1,70 +1,87 @@
 package semrel
 
 import (
+	"fmt"
 	"strings"
 	"testing"
 
 	"github.com/Masterminds/semver"
+	"github.com/go-semantic-release/semantic-release/pkg/config"
 )
 
-func TestCaluclateChange(t *testing.T) {
+func TestCalculateChange(t *testing.T) {
 	commits := []*Commit{
 		{SHA: "a", Change: Change{true, false, false}},
 		{SHA: "b", Change: Change{false, true, false}},
 		{SHA: "c", Change: Change{false, false, true}},
 	}
-	change := CaluclateChange(commits, &Release{})
+	change := CalculateChange(commits, &Release{})
 	if !change.Major || !change.Minor || !change.Patch {
 		t.Fail()
 	}
-	change = CaluclateChange(commits, &Release{SHA: "a"})
+	change = CalculateChange(commits, &Release{SHA: "a"})
 	if change.Major || change.Minor || change.Patch {
 		t.Fail()
 	}
 	version, _ := semver.NewVersion("1.0.0")
-	newVersion := GetNewVersion(commits, &Release{SHA: "b", Version: version})
+	newVersion := GetNewVersion(&config.Config{}, commits, &Release{SHA: "b", Version: version})
 	if newVersion.String() != "2.0.0" {
 		t.Fail()
 	}
 }
 
 func TestApplyChange(t *testing.T) {
-	version, _ := semver.NewVersion("1.0.0")
-	newVersion := ApplyChange(version, Change{false, false, false})
-	if newVersion != nil {
-		t.Fail()
-	}
-	newVersion = ApplyChange(version, Change{false, false, true})
-	if newVersion.String() != "1.0.1" {
-		t.Fail()
-	}
-	newVersion = ApplyChange(version, Change{false, true, true})
-	if newVersion.String() != "1.1.0" {
-		t.Fail()
-	}
-	newVersion = ApplyChange(version, Change{true, true, true})
-	if newVersion.String() != "2.0.0" {
-		t.Fail()
-	}
-	version, _ = semver.NewVersion("0.1.0")
-	newVersion = ApplyChange(version, Change{})
-	if newVersion.String() != "1.0.0" {
-		t.Fail()
-	}
-	version, _ = semver.NewVersion("2.0.0-beta")
-	newVersion = ApplyChange(version, Change{true, true, true})
-	if newVersion.String() != "2.0.0-beta.1" {
-		t.Fail()
-	}
-	version, _ = semver.NewVersion("2.0.0-beta.2")
-	newVersion = ApplyChange(version, Change{true, true, true})
-	if newVersion.String() != "2.0.0-beta.3" {
-		t.Fail()
+	NoChange := Change{false, false, false}
+	PatchChange := Change{false, false, true}
+	MinorChange := Change{false, true, true}
+	MajorChange := Change{true, true, true}
+
+	testCases := []struct {
+		currentVersion                  string
+		change                          Change
+		expectedVersion                 string
+		allowInitialDevelopmentVersions bool
+	}{
+		// No Previous Releases
+		{"0.0.0", NoChange, "1.0.0", false},
+		{"0.0.0", PatchChange, "1.0.0", false},
+		{"0.0.0", MinorChange, "1.0.0", false},
+		{"0.0.0", MajorChange, "1.0.0", false},
+
+		{"0.0.0", NoChange, "0.1.0", true},
```

Comment:
I think this case is incorrect because it creates confusing behaviour. 
This makes it seem like no changes will always create a release. And it will only happen the first time?

Reply:
semantic-release should always create the first release after it is configured, the JavaScript version has the same behavior: https://github.com/semantic-release/semantic-release/blob/master/lib/get-next-version.js#L30
Also, the server spec suggests starting with 0.1.0: https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

Reply:
Thanks for the detailed Reply

Reply:
But does this mean our default behaviour is still incorrect?